p2p/src/net/types/mod.rs

```diff
@@ -49,9 +55,8 @@ pub struct PeerInfo<T: NetworkingService> {
     /// User agent of the peer
     pub agent: Option<String>,
 
-    // TODO: protocolid must not generic!
-    /// List of supported protocols
-    pub protocols: Vec<T::ProtocolId>,
+    /// A set of supported protocols.
+    pub protocols: HashSet<Protocol>,
```

Comment:
Please use a BTreeSet. Usually the number of supported protocols is very small, hence, no need for a hash set. Generally, we try to avoid HashMap and HashSet unless we definitely need them because that typically causes confusion in hash function uses since we're doing blockchain and hashing is everywhere.